File: backend/TenderGo.Models/Requests/AdminTenderSearchRequest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using TenderGo.Models.ENUMs;

namespace TenderGo.Models.Requests
{
    public class AdminTenderSearchRequest:PagedSearchRequest
    {
        public string? SearchTerm { get; set; }
        public int? CategoryId { get; set; }
        public int? LocationId { get; set; }
        public TenderStatus? Status { get; set; }
    }
}

File: UI/tendergo_desktop/lib/shared/models/requests/admin_tender_search_request.dart
import 'package:tendergo/shared/models/enums/tenderstatus.dart';
import 'package:tendergo/shared/models/requests/paged_search_request.dart';

class AdminTenderSearchRequest extends PagedSearchRequest {
  final String? searchTerm;
  final TenderStatus? status;

  AdminTenderSearchRequest({
    this.searchTerm,
    this.status,
    int page = 1,
    int pageSize = 3, 
  }) : super(page: page, pageSize: pageSize);

  @override
  Map<String, dynamic> toJson() {
    return {
      'Page': page,
      'PageSize': pageSize,
      if (searchTerm != null && searchTerm!.trim().isNotEmpty) 'SearchTerm': searchTerm!.trim(),
      if (status != null) 'Status': status!.index + 1,
    };
  }
}